### PR TITLE
6LoWPAN: Assorted Refactorings

### PR DESCRIPTION
**Pull request by kkrentz**
This PR does some improvements to `sicslowpan.c`.

### CODE REVIEW

File: os/net/ipv6/sicslowpan.c [hunk 1]
```diff
@@ -689,6 +686,8 @@ compress_hdr_iphc(void)
   uint8_t tmp, iphc0, iphc1, *next_hdr, *next_nhc;
   int ext_hdr_len;
   struct uip_udp_hdr *udp_buf;
+  struct sicslowpan_addr_context *source_context;
+  struct sicslowpan_addr_context *destination_context;
```
Comment (pjonsson) [inline os/net/ipv6/sicslowpan.c:688]: Every platform supports C99 now, so declare variables closer to their initialization.

File: os/net/ipv6/sicslowpan.c [hunk 2]
```diff
@@ -893,15 +892,16 @@ compress_hdr_iphc(linkaddr_t *link_destaddr)
       /* compession compare with link adress (destination) */
 
       iphc1 |= compress_addr_64(SICSLOWPAN_IPHC_DAM_BIT,
-                                &UIP_IP_BUF->destipaddr,
-                                (uip_lladdr_t *)link_destaddr);
+          &UIP_IP_BUF->destipaddr,
+          (const uip_lladdr_t *)packetbuf_addr(PACKETBUF_ADDR_RECEIVER));
```
Comment (pjonsson) [inline os/net/ipv6/sicslowpan.c:891]: Shouldn't need to cast to const in these places.

**Comment by pjonsson** [comment]
Unless there are dependencies between these commits:

I'm guessing the NETSTACK_CONF_BRIDGE_MODE commit, along with the last two commits are trivial to review (the parentheses for macros are obviously an improvement, the two other commits are also correct so it's a question of whether they should be merged or not). Putting them in a separate PR should get them merged quicker.

The uncomp_hdr_len initialization should be fairly easy to review, so might be worth putting that in a separate PR too.

File: os/net/ipv6/sicslowpan.c (hunk 2 above)
Comment (kkrentz) [reply os/net/ipv6/sicslowpan.c:891]: Was necessary because uip uses a different type ...

**Pull request closed without merging**